### PR TITLE
task/WG-232-React-Listing-UI

### PR DESCRIPTION
**Pull request by sophia-massie**
## PR Status: ##

* [X] Ready.
* [ ] Work in Progress.
* [ ] Hold.

## Related Jira tickets: ##

* [WG-232](https://tacc-main.atlassian.net/browse/WG-232)
* [WG-233](https://tacc-main.atlassian.net/browse/WG-233)

## Summary of Changes: ##
Uses styles from Hazmapper angular to create Project Listing and cleans up the main project listing page

## Testing Steps: ##
1. Run nvm use 20 and npm ci
2. Run react front end and go to http://localhost:4200/. Check the UI so that it matches the Jira expectations

## UI Photos:
Screenshot 2024-11-14 at 2 47 55 PM

Error message:
Screenshot 2024-11-14 at 2 48 23 PM

No projects message:
Screenshot 2024-11-14 at 2 48 55 PM

### CODE REVIEW

**Review by nathanfranklin — CHANGES_REQUESTED** [review]
Awesome 👍   left some comments.

File: react/src/components/utils/EmptyTablePlaceholder.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { SectionMessage } from '../../core-components';
```
Comment (nathanfranklin) [inline react/src/components/utils/EmptyTablePlaceholder.tsx:1]: here are some comments for this component, but depending on https://github.com/TACC-Cloud/hazmapper/pull/268#discussion_r1805297011, they might not be needed.  so you might ignore this if EmptyPLaceholder is replaced by something else:

* do we want put this in `src/component/utils`?  is that spot for smaller components that will be reused by things in `src/components`?  if it is very Projects or ProjectListing and only used there, we could stick under src/components/Projects/
* the no children handling of "No data available" is never used. drop?

File: react/src/components/Projects/ProjectListing.tsx [hunk 2]
```diff
@@ -16,51 +18,79 @@ export const ProjectListing: React.FC = () => {
...
+  const { data, isLoading, isError, error } =
+    useProjectsWithDesignSafeInformation();
 
   if (isLoading) {
     return <LoadingSpinner />;
   }
 
   if (isError) {
-    return <h4>Unable to retrieve projects</h4>;
+    return (
```
Comment (sophia-massie) [inline react/src/components/Projects/ProjectListing.tsx:29]: @nathanfranklin If I move the error message out of the table, the Create New Map button never renders. Should I have a header row here like in return statement if there is an error loading maps?

Comment (nathanfranklin) [reply react/src/components/Projects/ProjectListing.tsx:29]: Our `error` covers something wrong with the DS project listing and/or Geoapi project. So, I think it's okay to just show an error message and not have buttons or listings (like the Create New Map button).

Usually, if one of those two endpoints returns an error, then it's a sign that they (Geoapi or Design Safe) are down/offline.

File: react/src/components/Projects/ProjectListing.tsx [hunk 3]
```diff
@@ -22,45 +24,85 @@ export const ProjectListing: React.FC = () => {
...
-                {proj.ds_project?.value.title}
-              </td>
-              <td>
-                <Button>
-                  <Icon name="edit-document"></Icon>
+    <div className={styles.root}>
+      <div className={styles.projectList}>
+        <table>
+          <thead className={styles.projectHeader}>
+            <tr>
```
[the remark this answers is not in the document]
Comment (sophia-massie) [reply react/src/components/Projects/ProjectListing.tsx]: Updated this so that it uses the colors like hazmapper

**Comment by sophia-massie** [comment]
Ready for another review

**Pull request merged**